ggml-cuda.cu

```diff
@@ -119,6 +119,24 @@ int ggml_cuda_get_device() {
     return id;
 }
 
+// ggml_cuda_host_malloc
+static inline cudaError_t ggml_cuda_device_malloc(void ** ptr, size_t size, int device) {
+#if defined(GGML_USE_HIPBLAS)
+#if defined(GGML_HIP_UMA)
+    auto res = hipMallocManaged(ptr, size);
+    if (res == hipSuccess) {
+        // if error we "need" to know why...
```

Comment:
Well it is what I want... may be need some explanation:

The function `ggml_cuda_device_malloc` is design to alloc "GPU" memory and return error code if the allocation failed.

Now with UMA memory allocation we need:
 - allocate the memory with `hipMallocManaged`
By default the memory is config with "Fine-grained" so the memory is coherent from RAM&VRAM this desactivate the GPU cache and made it slow. The other config is `Coarse-grained` that way the cache can be use, and we need to sync for use it on CPU . Because it is use like VRAM it is already sync.
 - config is done with call `hipMemAdvise` on the new memory

Now the choice a made: 
- if the alloc failed I retrun the error and let the caller deside what to do. 
- if the alloc sucess I configure it, and if config failed I stop the execution. 

Because:
- AMD stat that failure on `hipMemAdvise` report a error on parameter, so if I don't make error it can't failed ;) 
- Return the error in that case will need to find what to do with the allocated memory.
This 2 elements made me prefer to stop in case of memory config error.

the comment 
```
// if error we "need" to know why...
```
is for `hipMemAdvise` ... not `hipMallocManaged` and explain the use of  CUDA_CHECK with it. May be not clear.

As I say I made a choice for that so if you thing of better codding please tell me, it is my first PR here, happy to learn.

Reply:
The implementation is good, we just need more testing to ensure that it doesn't cause regressions in some cases.

Reply:
Thanks. The comment makes more sense knowing it refers to the line after, not the scope. I also think the code is fine, but if you wish, you could clarify the comment for the next person wondering the same thing as me. I'm not a reviewer here, by the way, so feel free to ignore.